### PR TITLE
Fix deprecations

### PR DESCRIPTION
**Pull request by abhijithch**
Fixes deprecations on `Julia v0.5`.

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -1,4 +1,5 @@
 using BinDeps
+using Compat
```
Comment (tkelman) [inline deps/build.jl]: set a minimum version in `REQUIRE` - you need to do this, every time

File: deps/build.jl [hunk 2]
```diff
@@ -65,8 +66,8 @@ provides(SimpleBuild,
     end),libipopt, os = :Unix)
 
 # OS X
-@osx_only begin
-    using Homebrew
+@compat @compat if is_apple()    
+	using Homebrew
```
Comment (tkelman) [inline deps/build.jl]: spaces, not tabs

File: src/Ipopt.jl [hunk 3]
```diff
@@ -204,7 +205,7 @@ function freeProblem(prob::IpoptProblem)
 end
 
 
-function addOption(prob::IpoptProblem, keyword::ASCIIString, value::ASCIIString)
+function addOption(prob::IpoptProblem, keyword::String, value::String)
```
Comment (tkelman) [inline src/Ipopt.jl]: if the underlying library can't handle unicode, this needs to be checked that it's ascii

Comment (abhijithch) [inline src/Ipopt.jl]: Can we use `Compat.ASCIIString` instead of `String`?

Comment (tkelman) [inline src/Ipopt.jl]: that won't do any verification on 0.5 so doesn't address the issue

File: deps/build.jl [hunk 4]
```diff
@@ -65,7 +66,7 @@ provides(SimpleBuild,
     end),libipopt, os = :Unix)
 
 # OS X
-@osx_only begin
+@compat @compat if is_apple()
```
Comment (tkelman) [inline deps/build.jl]: `@compat` not needed once, definitely not needed twice

File: src/Ipopt.jl [hunk 5]
```diff
@@ -204,9 +205,12 @@ function freeProblem(prob::IpoptProblem)
...
     #/** Function for adding a string option.  Returns FALSE the option
     # *  could not be set (e.g., if keyword is unknown) */
+    if !isascii(keyword)
```
Comment (tkelman) [inline src/Ipopt.jl]: and `value` ?

File: src/Ipopt.jl [hunk 6]
```diff
@@ -240,7 +250,7 @@ function addOption(prob::IpoptProblem, keyword::ASCIIString, value::Integer)
 end
 
 
-function openOutputFile(prob::IpoptProblem, file_name::ASCIIString, print_level::Int)
+function openOutputFile(prob::IpoptProblem, file_name::String, print_level::Int)
```
Comment (tkelman) [inline src/Ipopt.jl]: verification?

File: src/Ipopt.jl (hunk 3 above)
Comment (abhijithch) [inline src/Ipopt.jl]: Have updated the PR based on the comments. I think something is wrong with the `Julia` nightly build, issue posted [here](https://github.com/travis-ci/travis-ci/issues/6327).

**Comment by mlubin** [comment]
@tkelman, I'll let you merge when this is ready

**Pull request merged**